service/src/lib.rs

```diff
@@ -99,8 +101,30 @@ where
 		block_announce_validator,
 	);
 
-	let (polkadot_future, polkadot_task_manager) =
-		polkadot_collator::start_collator(builder, para_id, collator_key, polkadot_config)?;
+	let (polkadot_future, polkadot_task_manager) = if test {
+		let (task_manager, client, handles, _network, _rpc_handlers) = polkadot_test_service::polkadot_test_new_full(
```

Comment:
This should not be done here in this crate.

In the future I also don't want to use the rococo-collator for testing. We need a collator that is just used for testing. We really need to rework all this.

So, please move this functionality for now into the `rococo-collator`.